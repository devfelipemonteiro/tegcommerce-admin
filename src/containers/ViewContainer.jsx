import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import View from '../components/View';
import HeaderContainer from './HeaderContainer';
import FooterContainer from './FooterContainer';

class ViewContainer extends PureComponent {
  render() {
    const { children, ...others } = this.props;
    return (
      <View header={<HeaderContainer />} footer={<FooterContainer />} {...others}>
        {children}
      </View>
    );
  }
}

ViewContainer.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf(PropTypes.node)]),
};

ViewContainer.defaultProps = {
  children: undefined,
};

export default connect()(ViewContainer);
