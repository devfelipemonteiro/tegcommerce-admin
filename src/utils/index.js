import { Iterable, fromJS } from 'immutable';
import { createBrowserHistory } from 'history';

export const asImmutable = obj => (Iterable.isIterable(obj) ? obj : fromJS(obj));

export const emptyMap = asImmutable({});

export const getHistory = () => {
  window.browserHistory = window.browserHistory || createBrowserHistory();

  return window.browserHistory;
};

export const isDevEnv = () => {
  const nodeEnv = process.env.NODE_ENV;

  return nodeEnv !== 'production' && nodeEnv !== 'test';
};

export const debounce = (func, wait, immediate) => {
  let timeout;
  return function _debounce(...args) {
    const later = () => {
      timeout = null;
      if (!immediate) {
        func.apply(this, args);
      }
    };
    const callNow = immediate && !timeout;
    clearTimeout(timeout);
    timeout = setTimeout(later, wait);
    if (callNow) {
      func.apply(this, args);
    }
  };
};
