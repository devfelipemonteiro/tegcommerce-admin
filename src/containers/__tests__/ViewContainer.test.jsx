import React from 'react';
import { shallow } from 'enzyme';
import ViewContainer from '../ViewContainer';
import { createReduxTestStore } from '../../utils/test';

const minimalProps = {
  title: 'View Title'
};

describe('<ViewContainer /> Tests', () => {
  it('should render without exploding', done => {
    const testStore = createReduxTestStore();
    const wrapper = shallow(<ViewContainer store={testStore} />);
    expect(wrapper).toHaveLength(1);
    done();
  });

  it('should render correctly with minimal props', done => {
    const testStore = createReduxTestStore();
    const wrapper = shallow(
      <ViewContainer store={testStore} {...minimalProps}>
        <div>content</div>
      </ViewContainer>
    );
    expect(wrapper.dive().shallow()).toMatchSnapshot();
    done();
  });

  it('should render correctly with header prop', done => {
    const testStore = createReduxTestStore();
    const wrapper = shallow(
      <ViewContainer store={testStore} header={<div className="header" />} {...minimalProps} />
    );
    expect(wrapper.dive().shallow()).toMatchSnapshot();
    done();
  });

  it('should render correctly with footer prop', done => {
    const testStore = createReduxTestStore();
    const wrapper = shallow(
      <ViewContainer store={testStore} footer={<div className="footer" />} {...minimalProps} />
    );
    expect(wrapper.dive().shallow()).toMatchSnapshot();
    done();
  });
});
