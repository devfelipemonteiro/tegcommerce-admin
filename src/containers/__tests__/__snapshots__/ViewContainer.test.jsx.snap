// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewContainer /> Tests should render correctly with footer prop 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <div>
      <div
        className="headerContainer"
      >
        <Connect(HeaderContainer) />
      </div>
      <main
        className="mainContainer"
      >
        <h2
          className="viewTitle"
        >
          View Title
        </h2>
      </main>
      <div
        className="footerContainer"
      >
        <div
          className="footer"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<ViewContainer /> Tests should render correctly with header prop 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <div>
      <div
        className="headerContainer"
      >
        <div
          className="header"
        />
      </div>
      <main
        className="mainContainer"
      >
        <h2
          className="viewTitle"
        >
          View Title
        </h2>
      </main>
      <div
        className="footerContainer"
      >
        <FooterContainer />
      </div>
    </div>
  </div>
</div>
`;

exports[`<ViewContainer /> Tests should render correctly with minimal props 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <div>
      <div
        className="headerContainer"
      >
        <Connect(HeaderContainer) />
      </div>
      <main
        className="mainContainer"
      >
        <h2
          className="viewTitle"
        >
          View Title
        </h2>
        <div>
          content
        </div>
      </main>
      <div
        className="footerContainer"
      >
        <FooterContainer />
      </div>
    </div>
  </div>
</div>
`;
